apps/root/src/pages/earn-access-now/components/elegibility-confirmation/index.tsx

```diff
@@ -74,37 +78,39 @@ const STATUS_MAP: Record<ElegibilityConfirmationStatus, React.ComponentType<Eleg
 };
 
 const StatusWithElipse: ElegibilityConfirmationStatus[] = [
-  ElegibilityConfirmationStatus.NOT_ELIGIBLE,
   ElegibilityConfirmationStatus.ELIGIBLE,
   ElegibilityConfirmationStatus.NEEDS_SIGNATURE,
   ElegibilityConfirmationStatus.OWNERSHIP_CONFIRMED,
 ];
 
 const StatusWithSuccessAnimation: ElegibilityConfirmationStatus[] = [
-  ElegibilityConfirmationStatus.NOT_ELIGIBLE,
   ElegibilityConfirmationStatus.ELIGIBLE,
   ElegibilityConfirmationStatus.NEEDS_SIGNATURE,
 ];
 
 const ElegibilityConfirmation = () => {
-  const [status, setStatus] = React.useState(ElegibilityConfirmationStatus.LOADING);
-  const wallets = useWallets();
-  const elegibleWallets: Address[] = wallets.map((wallet) => wallet.address);
+  const { fetchElegibilityAchievements, elegibilityStatus, setElegibilityStatus, elegibleWallets } =
+    useElegibilityCriteria();
+  const wallets = useWallets().map((wallet) => wallet.address);
```

Comment:
map is always going to create a new object, it would be better to parse this map on the useEffect directlyt